src/Entity.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace UKFast\SDK;
+
+abstract class Entity
+{
+    /**
+     * @var array
+     */
+    protected $attributes = [];
+
+    /**
+     * An array of read only attributes
+     * @var array
+     */
+    protected $readOnly = [];
```

Comment:
If we are only using the `readOnly` array to remove columns from the JSON representation. Maybe we could name it `hidden` like its Eloquent equivalent. Just an opinion.

Reply:
The idea behind this is that we have a list of properties we know not to send on post/patch endpoints. It's similar-ish to eloquent's hidden, but I think this shows its intent more clearly.

Reply:
Is it possible to add an array which restricts what can be filled.

Reply:
I'm happy to to an array which restricts what can be filled, but I think that'd be better off in a separate PR to keep this one smaller